src/id.rs

```diff
@@ -96,21 +98,16 @@ impl parsec::SecretId for FullId {
     }
 
     fn sign_detached(&self, data: &[u8]) -> <Self::PublicId as parsec::PublicId>::Signature {
-        self.signing_private_key().sign_detached(data)
+        self.sign(data)
     }
 
-    fn encrypt<M: AsRef<[u8]>>(&self, to: &Self::PublicId, msg: M) -> Option<Vec<u8>> {
-        let shared_secret = self
-            .encrypting_private_key()
-            .shared_secret(to.encrypting_public_key());
-        shared_secret.encrypt_bytes(msg.as_ref()).ok()
+    fn encrypt<M: AsRef<[u8]>>(&self, to: &Self::PublicId, plaintext: M) -> Option<Vec<u8>> {
```

Comment:
Yes, pity this trait demands type erasure.  Later we can fix that and use a generic for encrypted data types. That will be  a PARSEC update though.

Reply:
Why is that a pity?

Reply:
Because it forces us to serialise the `Ciphertext` type which is additional work that wouldn't be necessary otherwise.

Reply:
It is a low occurring event so performance is not a problem, also it will be serialized on the wire so probably not make much gain.
On the other hand, making it generic means that parsec code & DKG has to cater for generic parameter where it only really care about a blob of data from this encrypt function.

So I'm not sure it will be a worthwhile trade off in the end but we'll see.

Reply:
I see type erasure always as a pity ;-) Here I am also still cool to merge though. In this case for parsec at least we can go back to Ciphertext and validate before trying to decrypt. So the overhead is on both sides of the encrypt/decrypt setup here.